time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/NodeCalcEvaluator.java

```diff
@@ -9,13 +9,23 @@
 import com.powsybl.timeseries.DoubleMultiPoint;
 import org.apache.commons.lang3.tuple.Pair;
 
+import java.util.IdentityHashMap;
+import java.util.Map;
+
 /**
  * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
  */
 public class NodeCalcEvaluator implements NodeCalcVisitor<Double, DoubleMultiPoint> {
 
+    private Map<NodeCalc, Double> cache;
```

Comment:
So now the NodeCalcEvaluator is not stateless anymore. Maybe we could have use the visitor context for that. So instead of DoubleMultiPoint could be a 

```java
class EvalContext {
      DoubleMultiPoint point;
      Map<NodeCalc, Double> cache;
}
```

Reply:
modified